src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -129,7 +129,7 @@ public sealed class DiagnosticDescriptor : IEquatable<DiagnosticDescriptor>
             string id,
             LocalizableString title,
             LocalizableString messageFormat,
-            string category,
+            LocalizableString category,
```

Comment:
You need to invoke `MakeExceptionSafe` for category initialization below to ensure that exceptions from it don't go unhandled.
Additionally, there are couple of places in AnalyzerManager where we hook up handlers for reporting diagnostics for exceptions from all localizable string fields of the descriptor. You should add handler registration/un-registration code for Desciptor.Category in there. 
Additionally, we also have a test in CodeAnalysisTest project where we check that localizable strings of descriptor are exception safe. We should add to those tests to check for exceptions from LocalizableString for category.